modules/custom/lua/custom_HNM_system.lua

```diff
@@ -61,8 +67,14 @@ hnmSystem:addOverride("xi.zones.Dragons_Aery.mobs.Fafnir.onMobDespawn", function
 end)
 
 hnmSystem:addOverride("xi.zones.Dragons_Aery.npcs.qm0.onTrade", function(player, npc, trade)
-    if not GetMobByID(dragonsAeryID.mob.FAFNIR):isSpawned() and not GetMobByID(dragonsAeryID.mob.NIDHOGG):isSpawned() then
-        if npcUtil.tradeHasExactly(trade, 3340) and npcUtil.popFromQM(player, npc, dragonsAeryID.mob.NIDHOGG) then
+    if
+        not GetMobByID(dragonsAeryID.mob.FAFNIR):isSpawned() and
+        not GetMobByID(dragonsAeryID.mob.NIDHOGG):isSpawned()
+    then
+        if
+            npcUtil.tradeHasExactly(trade, 3340) and
```

Comment:
Can these trade items get added to items.lua?

Also, the nested condition could be simplified to just one, since there's only one single path